google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -687,3 +663,15 @@ def _connect_with_pytds(
         return pytds.connect(
             ip_address, database=db, user=user, password=passwd, sock=sock, **kwargs
         )
+
+    async def close(self) -> None:
+        """Cleanup function to make sure ClientSession is closed and tasks have
+        finished to have a graceful exit.
+        """
+        logger.debug("Waiting for _current to be cancelled")
+        self._current.cancel()
+        logger.debug("Waiting for _next to be cancelled")
+        self._next.cancel()
+        if not self._client_session.closed:
```

Comment:
Since we implement ClientSession explicitly without using it as a context manager there are times when it is not closed. However, no error occurs if you try to close an already closed ClientSession so I've removed the if statement.